src/infinite-scroll/index.ts

```diff
@@ -183,7 +278,7 @@ class InfiniteScroll {
 
   setProducts = (products?: Core.Store.ProductWithMetadata[]) => {
     let items;
-    if (products) {
+    if (products.length && this.state.items.length) {
```

Comment:
Is there a chance that `products` will be undefined? If so, lets ensure that it exists before operating on it.

Reply:
Products will never be undefined the way we're calling it, so I updated the definition. The only reason I can think to check if it's undefined is because a user might call it with undefined which begs the question whether we should be doing checks like that everywhere

Reply:
@Vic-Dev Got it, thanks for the context.